src/py/flwr/common/logger.py

```diff
@@ -290,3 +298,58 @@ def restore_output() -> None:
     sys.stdout = sys.__stdout__
     sys.stderr = sys.__stderr__
     console_handler.stream = sys.stdout
+
+
+def _log_uploader(
+    log_queue: Queue[Optional[str]], node_id: int, run_id: int, stub: DriverStub
+) -> None:
+    """Upload logs to the SuperLink."""
+    exit_flag = False
+    node = Node(node_id=node_id, anonymous=False)
+    while True:
+        msgs: list[str] = []
+        # Fetch all messages from the queue
+        try:
+            while True:
+                msg = log_queue.get_nowait()
+                # Quit the loops if the returned message is `None`
+                # This is a signal that the run has finished
+                if msg is None:
+                    exit_flag = True
+                    break
+                msgs.append(msg)
+        except Empty:
+            pass
+
+        # Upload if any logs
+        if msgs:
+            req = PushLogsRequest(
+                node=node,
+                run_id=run_id,
+                logs=msgs,
+            )
+            stub.PushLogs(req)
```

Comment:
This needs error handling. We shouldn't crash the log uploader if a single request fails.

Reply:
Sure, that makes sense.